src/digests.rs

```diff
@@ -351,13 +348,12 @@ impl<'a, T: TryFrom<u64>> MultihashRefGeneric<'a, T> {
     /// let mh2 = MultihashRef::from_slice(&mh).unwrap();
     /// assert_eq!(mh2.algorithm(), Code::Sha2_256);
     /// ```
-    pub fn algorithm(&self) -> T
-    where
-        <T as TryFrom<u64>>::Error: std::fmt::Debug,
-    {
+    pub fn algorithm(&self) -> T {
         let (rawcode, _bytes) =
             varint_decode::u64(&self.bytes).expect("multihash is known to be valid algorithm");
-        T::try_from(rawcode).expect("multihash is known to be a valid algorithm")
+        T::try_from(rawcode)
```

Comment:
Oh wait, couldn't this just be:

```rust
T::try_from(rawcode).unwrap_or_else(|_| {
    panic!("Should not occur as multihash is known to be valid")
})
```

Reply:
:) Good call, what you wrote expresses intent and effect a bit better. Will update.